reader-osm/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -431,7 +430,7 @@ public void processRelation(ReaderRelation relation) {
         Map<Long, TurnCostTableEntry> entries = new LinkedHashMap<>();
 
         for (FlagEncoder encoder : encodingManager.fetchEdgeEncoders()) {
-            if (encoder.acceptsRelation(turnRelation)) {
+            if ((turnRelation == null) || ((turnRelation != null) && encoder.acceptsRelation(turnRelation))) {
```

Comment:
We can remove this null check (we already check for null after `createTurnRelation` and the 'default' is 'nothing is null').

Reply:
If I remove the test `turnRelation == null`, `OSMReaderTest#testTurnFlagCombination` fails.
There are 2 entries expected from the call:
```
Collection<OSMReader.TurnCostTableEntry> entries = reader.analyzeTurnRelation(null);
// we expect two different turnCost entries
assertEquals(2, entries.size());
```
Without the test, `entries` is empty, so the assert fails.

Reply:
Ok, yes, but `turnRelation` should not be null here in the first place. In fact it can never be null in production (because of the null check right after `createTurnRelation`. In `OSMReaderTest#testTurnFlagCombination` we explicitly pass null, and in fact this test uses some customized `analyzeTurnRelation` method that does not even us the `turnRelation`. So I think we should remove the null check and fix the test by passing just some `OSMTurnRelation` instance in `testTurnFlagCombination` instead of null.

Reply:
Yeah, thanks for your work! This looks good.